src/sync/sync_session.cpp

```diff
@@ -514,6 +562,47 @@ bool SyncSession::wait_for_upload_completion_blocking()
     return false;
 }
 
+uint64_t SyncSession::register_progress_notifier(std::function<SyncProgressNotifierCallback> notifier,
+                                                 NotifierType direction,
+                                                 bool is_streaming)
+{
+    std::function<void()> invocation;
+    uint64_t token_value = 0;
+    {
+        std::lock_guard<std::mutex> lock(m_progress_notifier_mutex);
+        token_value = m_progress_notifier_token;
+        m_progress_notifier_token++;
+        auto current_transferrable = (direction == NotifierType::download ? m_current_downloadable : m_current_uploadable);
+        auto current_transferred = (direction == NotifierType::download ? m_current_downloaded : m_current_uploaded);
+        // Always invoke the notifier immediately, to give the user up-to-date info.
+        if (!current_transferrable) {
```

Comment:
This used to be testing for the absence of information. Now it's testing for the value 0. Is that intentional?